cmd/local-gadget/trace/network.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	commontrace "github.com/inspektor-gadget/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/networktracer"
+	networkTracer "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/tracer"
+	networkTypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/types"
+	localgadgetmanager "github.com/inspektor-gadget/inspektor-gadget/pkg/local-gadget-manager"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+func newNetworkCmd() *cobra.Command {
+	var commonFlags utils.CommonFlags
+
+	// The network gadget works in a different way than most gadgets: It
+	// attaches a new eBPF program to each container when it's created instead
+	// of using an eBPF map with the mount namespaces IDs to filter the events.
+	// For this reason we can't use the TraceGadget implementation here.
+	runCmd := func(*cobra.Command, []string) error {
+		localGadgetManager, err := localgadgetmanager.NewManager(commonFlags.RuntimeConfigs)
+		if err != nil {
+			return commonutils.WrapInErrManagerInit(err)
+		}
+		defer localGadgetManager.Close()
+
+		// local-gadget is designed to trace containers, hence enable this column
+		cols := networkTypes.GetColumns()
+		col, _ := cols.GetColumn("container")
+		col.Visible = true
+
+		parser, err := commonutils.NewGadgetParserWithRuntimeInfo(&commonFlags.OutputConfig, cols)
+		if err != nil {
+			return commonutils.WrapInErrParserCreate(err)
+		}
+
+		// This callback is used by the ConnectionToContainerCollection to
+		// notify when containers are attached and detached, or any error during
+		// that operations. The actual events generated by the tracer will be
+		// retrieved using the Pop() method.
+		eventCallback := func(container *containercollection.Container, event networkTypes.Event) {
```

Comment:
Please update now that https://github.com/inspektor-gadget/inspektor-gadget/pull/1248 is merged.

Reply:
Done.